oak_functions/loader/src/server.rs

```diff
@@ -149,25 +148,54 @@ impl FixedSizeBodyPadder for Response {
     }
 }
 
+/// Trait for implementing extensions, to implement new native functionality.
+pub trait OakApiNativeExtension {
+    /// Similar to `invoke_index` in [`wasmi::Externals`], but may return a result to be
+    /// written into the memory of the `WasmState`.
+    fn invoke_index(
+        &self,
+        wasm_state: &WasmState,
+        args: wasmi::RuntimeArgs,
+    ) -> Result<Result<Option<ExtensionResult>, OakStatus>, wasmi::Trap>;
+
+    /// Similar to `resolve_func` in [`wasmi::ModuleImportResolver`], but returns a tuple with a
+    /// function index and a function signature to be converted into [`wasmi::FuncRef`].
```

Comment:
Could you clarify how to pick these indices, what they mean, what properties they should have, do they need to be sequential, etc.?